src/SessionTracker.php

```diff
@@ -281,27 +290,36 @@ protected function constructPayload($sessions)
     protected function deliverSessions()
     {
         $sessions = $this->getSessionCounts();
+
         $this->setSessionCounts([]);
-        if (count($sessions) == 0) {
+
+        if (!is_array($sessions) || count($sessions) === 0) {
             return;
         }
+
         if (!$this->config->shouldNotify()) {
             return;
         }
+
         $http = $this->config->getSessionClient();
-        $payload = $this->constructPayload($sessions);
-        $headers = [
-            'Bugsnag-Api-Key' => $this->config->getApiKey(),
-            'Bugsnag-Payload-Version' => self::$SESSION_PAYLOAD_VERSION,
-            'Bugsnag-Sent-At' => strftime('%Y-%m-%dT%H:%M:%S'),
+
+        $options = [
+            'json' => $this->constructPayload($sessions),
+            'headers' => [
+                'Bugsnag-Api-Key' => $this->config->getApiKey(),
+                'Bugsnag-Payload-Version' => self::$SESSION_PAYLOAD_VERSION,
+                'Bugsnag-Sent-At' => strftime('%Y-%m-%dT%H:%M:%S'),
+            ],
         ];
+
         $this->setLastSent();
 
         try {
-            $http->post('', [
-                'json' => $payload,
-                'headers' => $headers,
-            ]);
+            if (method_exists($http, 'request')) {
```

Comment:
This code should match https://github.com/bugsnag/bugsnag-php/commit/3264e0da5c4082337659870752cffa16a4e5920e#diff-add00846710b59ac0af2a07e1ede2747R209-R225. `method_exists` is technically wrong because it could detect something which is not a public method.

Reply:
Just making sure I understand — you mean check the `ClientInterface` specifically, rather than the `$http` object?

Reply:
Yes. This is because the interface tells you about the public API, rather than not necessarily public class methods.